kubetest/azure.go

```diff
@@ -0,0 +1,348 @@
+		return fmt.Errorf("Failed to generate ARM templates: %v", err)
+	}
+	return nil
+}
+
+func (c *Cluster) loadARMTemplates() error {
+	var err error
+	template, err := ioutil.ReadFile(path.Join(c.outputDir, "azuredeploy.json"))
+	if err != nil {
+		return fmt.Errorf("Error reading ARM template file: %v", err)
+	}
+	err = json.Unmarshal([]byte(template), &c.templateJSON)
+	if err != nil {
+		return fmt.Errorf("Error unmarshall template %v", err.Error())
+	}
+
+	parameters, err := ioutil.ReadFile(path.Join(c.outputDir, "azuredeploy.parameters.json"))
+	if err != nil {
+		return fmt.Errorf("Error reading ARM parameters file: %v", err)
+	}
+	err = json.Unmarshal([]byte(parameters), &c.parametersJSON)
+	if err != nil {
+		return fmt.Errorf("Error unmarshall parameters %v", err.Error())
+	}
+	return nil
+}
+
+func (c *Cluster) getARMClient(ctx context.Context) error {
+	env, err := azure.EnvironmentFromName("AzurePublicCloud")
+	var client *AzureClient
+	if client, err = getAzureClient(env,
+		c.subscriptionId,
+		c.clientId,
+		c.tenantId,
+		c.clientSecret); err != nil {
+		return fmt.Errorf("Error trying to get Azure Client: %v", err)
+	}
+	c.azureClient = client
+	return nil
+}
+
+func (c *Cluster) createCluster() error {
+	var err error
+	err = c.getARMClient(c.ctx)
+	if err != nil {
+		return fmt.Errorf("Failed to generate ARM client: %v", err)
+	}
+	kubecfgDir, _ := ioutil.ReadDir(path.Join(c.outputDir, "kubeconfig"))
+	kubecfg := path.Join(c.outputDir, "kubeconfig", kubecfgDir[0].Name())
+	fmt.Printf("Setting kubeconfig env variable: kubeconfig path: %v", kubecfg)
+	os.Setenv("KUBECONFIG", kubecfg)
+	_, err = c.azureClient.EnsureResourceGroup(c.ctx, c.resourceGroup, c.location, nil)
+	if err != nil {
+		return fmt.Errorf("Could not ensure resource group: %v", err)
+	}
+	if _, err := c.azureClient.ValidateDeployment(
+		c.ctx, c.resourceGroup, c.name, c.templateJSON, c.parametersJSON,
+	); err != nil {
+		return fmt.Errorf("Template invalid: %v", err)
+	}
+	if _, err := c.azureClient.DeployTemplate(
+		c.ctx,
+		c.resourceGroup,
+		c.name,
+		c.templateJSON,
+		c.parametersJSON); err != nil {
+		return fmt.Errorf("Cannot deploy: %v", err)
+	}
+	return nil
+
+}
+
+func (c Cluster) Up() error {
+	var err error
+	err = c.generateTemplate()
+	if err != nil {
+		return fmt.Errorf("Failed to generat apiModel: %v", err)
+	}
+	err = c.getAcsEngine(1)
+	if err != nil {
+		return fmt.Errorf("Failed to get ACS Engine binary: %v", err)
+	}
+	err = c.generateARMTemplates()
+	if err != nil {
+		return fmt.Errorf("Failed to generate ARM templates: %v", err)
+	}
+	err = c.loadARMTemplates()
+	if err != nil {
+		return fmt.Errorf("Error loading ARM templates: %v", err)
+	}
+	err = c.createCluster()
+	if err != nil {
+		return fmt.Errorf("Error creating cluster: %v", err)
+	}
+
+	return nil
+}
+
+func (c Cluster) Down() error {
+	_, err := c.azureClient.DeleteResourceGroup(c.ctx, c.resourceGroup)
```

Comment:
only delete when the group does not exist, in case the group has not been created